File: TICTACTOE/tictactoe.py
def display_board(board):
    for i in range(3):
        for j in range(3):
            print(f" {board[i][j]} ", end="")
            if j < 2:
                print("|", end="")
        print()
        if i < 2:
            print("-----------")

def check_win(board, player):
    for i in range(3):
        if all(board[i][j] == player for j in range(3)):
            return True
        if all(board[j][i] == player for j in range(3)):
            return True
    if all(board[i][i] == player for i in range(3)):
        return True
    if all(board[i][2 - i] == player for i in range(3)):
        return True
    return False

def main():
    board = [['1', '2', '3'], ['4', '5', '6'], ['7', '8', '9']]
    player_turn = 1
    total_moves = 0

    while total_moves < 9:
        display_board(board)

        if player_turn == 1:
            print("Player 1 (X), enter row and column (1-3): ")
        else:
            print("Player 2 (O), enter row and column (1-3): ")

        row, col = map(int, input().split())

        if row < 1 or row > 3 or col < 1 or col > 3 or board[row - 1][col - 1] == 'X' or board[row - 1][col - 1] == 'O':
            print("Invalid move. Try again.")
            continue

        if player_turn == 1:
            board[row - 1][col - 1] = 'X'
        else:
            board[row - 1][col - 1] = 'O'

        if check_win(board, 'X'):
            display_board(board)
            print("Player 1 (X) wins!")
            break
        elif check_win(board, 'O'):
            display_board(board)
            print("Player 2 (O) wins!")
            break

        player_turn = 3 - player_turn  # Switch players (1 to 2, and 2 to 1)
        total_moves += 1

    if total_moves == 9:
        display_board(board)
        print("It's a draw!")

if __name__ == "__main__":
    main()

File: TICTACTOE/tictactoe.js
//To run it, make sure you have Node.js installed, save the code in a .js file (e.g., ticTacToe.js), and execute it using the Node.js runtime.
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function displayBoard(board) {
    for (let i = 0; i < 3; i++) {
        for (let j = 0; j < 3; j++) {
            process.stdout.write(` ${board[i][j]} `);
            if (j < 2) process.stdout.write('|');
        }
        console.log();
        if (i < 2) console.log('-----------');
    }
}

function checkWin(board, player) {
    for (let i = 0; i < 3; i++) {
        if (board[i][0] === player && board[i][1] === player && board[i][2] === player) return true;
        if (board[0][i] === player && board[1][i] === player && board[2][i] === player) return true;
    }
    if (board[0][0] === player && board[1][1] === player && board[2][2] === player) return true;
    if (board[0][2] === player && board[1][1] === player && board[2][0] === player) return true;
    return false;
}

function main() {
    const board = [['1', '2', '3'], ['4', '5', '6'], ['7', '8', '9']];
    let playerTurn = 1;
    let totalMoves = 0;

    function getInput() {
        rl.question(`Player ${playerTurn} (X), enter row and column (1-3): `, (input) => {
            const [row, col] = input.split(' ').map(Number);

            if (row < 1 || row > 3 || col < 1 || col > 3 || board[row - 1][col - 1] === 'X' || board[row - 1][col - 1] === 'O') {
                console.log('Invalid move. Try again.');
                getInput();
                return;
            }

            if (playerTurn === 1) {
                board[row - 1][col - 1] = 'X';
            } else {
                board[row - 1][col - 1] = 'O';
            }

            if (checkWin(board, 'X')) {
                displayBoard(board);
                console.log('Player 1 (X) wins!');
                rl.close();
            } else if (checkWin(board, 'O')) {
                displayBoard(board);
                console.log('Player 2 (O) wins!');
                rl.close();
            } else {
                playerTurn = 3 - playerTurn; // Switch players (1 to 2, and 2 to 1)
                totalMoves++;

                if (totalMoves === 9) {
                    displayBoard(board);
                    console.log("It's a draw!");
                    rl.close();
                } else {
                    displayBoard(board);
                    getInput();
                }
            }
        });
    }

    displayBoard(board);
    getInput();
}

main();
